crypto/evp/evp_lib.c

```diff
@@ -232,8 +232,14 @@ int EVP_Cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
                const unsigned char *in, unsigned int inl)
 {
     if (ctx->cipher->prov != NULL) {
+        size_t outl = 0;         /* ignored */
+        int blocksize = EVP_CIPHER_CTX_block_size(ctx);
+
         if (ctx->cipher->ccipher != NULL)
-            return ctx->cipher->ccipher(ctx->provctx, out, in, (size_t)inl);
+            return
+                ctx->cipher->ccipher(ctx->provctx, out, &outl,
+                                     inl + (blocksize == 1 ? 0 : blocksize),
```

Comment:
Doesn't seem right for this one-shot operation. In the worst case scenario there will never get more than `inl` bytes out. This is different to the CipherUpdate and CipherFinal operations where previous calls can mean data is buffered so you can get more out than you put in. That isn't the case with EVP_Cipher. Its a shame that EVP_Cipher is undocumented.

Reply:
So EVP_Cipher() always acts like a stream cipher?  I.e. something like this should work?  This won't overwrite the stack?

``` C
unsigned char out[1];

EVP_Cipher(ctx, out, in, 1);
```

Reply:
Hmm. No. I'm talking rubbish. Ignore me.